src/Microsoft.AspNet.Mvc.DataAnnotations/Internal/MvcDataAnnotationsMvcOptionsSetup.cs

```diff
@@ -12,15 +15,26 @@ namespace Microsoft.AspNet.Mvc.DataAnnotations.Internal
     /// </summary>
     public class MvcDataAnnotationsMvcOptionsSetup : ConfigureOptions<MvcOptions>
     {
-        public MvcDataAnnotationsMvcOptionsSetup()
-            : base(ConfigureMvc)
+        public MvcDataAnnotationsMvcOptionsSetup(
+           IServiceProvider serviceProvider)
+            : base(options => ConfigureMvc(options, serviceProvider))
         {
         }
 
-        public static void ConfigureMvc(MvcOptions options)
+        public static void ConfigureMvc(
+            MvcOptions options,
+            IServiceProvider serviceProvider)
         {
+            var dataAnnotationLocalizationOptions =
+                serviceProvider.GetRequiredService<IOptions<MvcDataAnnotationsLocalizationOptions>>();
```

Comment:
It's odd that you bring in the `IServiceProvider` in the ctor, just to use it to further pull in additional services that may or may not exist. Is there a reason why we can't just specify in the ctor the dependencies this class needs then pass them through?